src/App.tsx

```diff
@@ -16,35 +17,90 @@ export const goodsFromServer = [
 ];
 
 export const App: React.FC = () => {
+  const [goodsList, setGoodsList] = useState<string[]>([...goodsFromServer]);
+  const [sortType, setSortType] = useState<string>('');
+  const [reversed, setReversed] = useState<boolean>(false);
+
+  const handleSort = (type: string): void => {
+    let sortedGoods = [...goodsList];
+
+    switch (type) {
+      case 'by name':
+        sortedGoods.sort();
+        break;
+      case 'by length':
+        sortedGoods.sort((good1, good2) => good1.length - good2.length);
+        break;
+      case 'reverse':
+        sortedGoods.reverse();
+        break;
+      default:
+        sortedGoods = [...goodsFromServer];
+    }
+
+    setSortType(type);
+    setGoodsList(sortedGoods);
+  };
```

Comment:
```suggestion
  const getPreparedGoods = (sortType: string, reversed: boolean): void => {
    let sortedGoods = [...goodsList];

    switch (type) {
      case 'by name':
        sortedGoods.sort();
        break;
      case 'by length':
        sortedGoods.sort((good1, good2) => good1.length - good2.length);
        break;
      default:
        sortedGoods = [...goodsFromServer];
    }
    
    if(reverse) {
      sortedGoods.reverse();
    }

   return sortedGoods
  };
```

and now you can just call this function and use the results everywhere
`const visibleGoods = getPreparedGoods(sortType, reversed)`